src/main/java/com/parksupark/soomjae/server/community/communitypost/service/CommunityPostService.java

```diff
@@ -30,75 +34,90 @@ public class CommunityPostService {
 
         Category category = null;
         if (communityPostRequest.getCategory() != null) {
             category = categoryRepository.findById(
-                            Long.parseLong(communityPostRequest.getCategory()))
-                    .orElseThrow(() -> new IllegalStateException(CATEGORY_NOT_FOUND));
+                    Long.parseLong(communityPostRequest.getCategory()))
+                .orElseThrow(() -> new IllegalStateException(CATEGORY_NOT_FOUND));
         }
         Location location = null;
         if (communityPostRequest.getLocation() != null) {
             location = locationRepository.findByCode(
-                            Long.parseLong(communityPostRequest.getLocation()))
-                    .orElseThrow(() -> new IllegalStateException(
-                            LocationConstant.LOCATION_NOT_FOUND));
+                    Long.parseLong(communityPostRequest.getLocation()))
+                .orElseThrow(() -> new IllegalStateException(
+                    LocationConstant.LOCATION_NOT_FOUND));
         }
         CommunityPost entity = communityPostRequest.toEntity(member, category, location);
 
         return communityPostRepository.save(entity).getId();
     }
 
     public CommunityPostListResponse readByFilter(Pageable pageable) {
-        Page<CommunityPost> postPage = communityPostRepository.findAll(pageable);
-        List<CommunityPostResponse> postResponseList = postPage.map(CommunityPostResponse::of)
-                .getContent();
-        return new CommunityPostListResponse(postResponseList);
+        List<CommunityPost> contents = communityPostRepository.findAll(pageable).getContent();
+        List<CommunityPostResponse> response = new ArrayList<>();
+        for (CommunityPost post : contents) {
+            long commentNum = commentRepository.countByPostTypeAndPostIdAndDeletedTimeIsNull(
+                COMMUNITY_POST_TYPE, post.getId());
+            CommunityPostResponse communityPostResponse = CommunityPostResponse.of(post,
+                commentNum);
+            response.add(communityPostResponse);
+        }
+        return CommunityPostListResponse.of(response);
     }
 
-
-    public CommunityPostResponse readBypostId(Long postId) {
+    public CommunityPostDetailResponse readBypostId(Long postId) {
         CommunityPost communityPost = communityPostRepository.findById(postId)
-                .orElseThrow(() -> new IllegalStateException(COMMUNITY_POST_NOT_FOUND));
-        return CommunityPostResponse.of(communityPost);
+            .orElseThrow(() -> new IllegalStateException(COMMUNITY_POST_NOT_FOUND));
+        List<CommentResponse> comments = commentRepository
+            .findByPostTypeAndPostIdAndDeletedTimeIsNull(
+                COMMUNITY_POST_TYPE, communityPost.getId()).stream().map(CommentResponse::of)
+            .toList();
+        return CommunityPostDetailResponse.of(communityPost, comments);
     }
 
     //페이징 기능 추 후 구현
```

Comment:
p3; 

```suggestion
    // TODO: 페이징 기능 추 후 구현
```

추후 구현해야 하는 내용은 앞에 TODO를 붙여야 한다는 convention을 만드는 것은 어떨까요? ide에서 어떤 todo가 남아있는지 추적도 가능합니다!

Reply:
좋습니다 노션 프로젝트 컨벤션에도 추가해놓겠습니다.